src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -433,3 +433,46 @@ g.test('limit,worse_than_default')
       t.shouldReject('OperationError', adapter.requestDevice({ requiredLimits }));
     }
   });
+
+g.test('always_returns_device')
+  .desc(
+    `
+    Test that if requestAdapter returns an adapter then requestDevice must return a device.
+
+    requestAdapter -> null = ok
+    requestAdapter -> adapter, requestDevice -> device (lost or not) = ok
+    requestAdapter -> adapter, requestDevice = null = Invalid: not spec compliant.
+
+    Note: requestDevice can throw for invalid parameters like requesting features not
+    in the adapter, reqesting limits not in the adapter, requesting limits larger than
+    the maximum for the adapter. Otherwise it does not throw.
+
+    Note: This is a regression test for a Chrome bug crbug.com/349062459
+    Checking that a requestDevice always return a device is checked in other tests above
+    but those tests have 'compatibilityMode: true' set for them by the API that getGPU
+    returns when the test suite is run in compatibility mode.
+
+    This test tries to force both compat and core separately so both code paths are
+    tested in the same browser configuration.
+  `
+  )
+  .params(u => u.combine('compatibilityMode', [false, true] as const))
+  .fn(async t => {
+    const { compatibilityMode } = t.params;
+    const gpu = getGPU(t.rec);
+    // MAINTENANCE_TODO: Remove this cast compatibilityMode is added.
+    const adapter = await gpu.requestAdapter({ compatibilityMode } as GPURequestAdapterOptions);
+    if (adapter) {
```

Comment:
not important, but note: I don't remember if any of the other WebGPU tests pass if requestAdapter returns null. If they would fail in that case, then really this test doesn't need to pass either.

Reply:
I think maybe I'm not understanding the your comment?

On a device that only supports compatibility mode, `requestAdapter()` should turn `null` whereas `requestAdapter({compatibilityMode: true})` will return a `GPUAdapter`.

Reply:
oh yes, I see. Never mind!